### DISCUSSION TITLE
Running hello container through podman-compose does not work, podman reports a problem with Pasta

### DISCUSSION THREAD

**Discussion by flixman** [containers/podman#23136] [Q&A] [answered]
I have a docker-compose.yml file that is very simple:
```yaml
services:
  hello:
    image: hello
```
if I run directly the hello image, with podman, all works. If I run it through podman-compose with the verbose mode, I get the following:

```sh
$ podman-compose --verbose run --rm hello
INFO:podman_compose:['podman', '--version', '']
INFO:podman_compose:using podman version: 5.1.1
INFO:podman_compose:podman pod create --name=pod_test --infra=false --share=
Error: adding pod to state: name "pod_test" is in use: pod already exists
INFO:podman_compose:exit code: 125
INFO:podman_compose:['podman', 'network', 'exists', 'test_default']
INFO:podman_compose:podman run --name=test_hello_tmp11366 --rm -i --pod=pod_test --label io.podman.compose.config-hash=5e36e74a382d7051c6fcb79d725d6225281563a8a995178f4030ef95c901375a --label io.podman.compose.project=test --label io.podman.compose.version=1.2.0 --label PODMAN_SYSTEMD_UNIT=podman-compose@test.service --label com.docker.compose.project=test --label com.docker.compose.project.working_dir=/test --label com.docker.compose.project.config_files=docker-compose.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=hello --network=test_default --network-alias=hello --tty hello
Error: setting up Pasta: pasta failed with exit code 1:
```

This happens inside of a podman container (so, I am trying to run podman-in-podman), created from this Dockerfile:
```dockerfile
FROM quay.io/podman/stable

RUN dnf install -y git nodejs && \
    dnf install -y python3-pip && \
    pip install podman-compose && \
    dnf clean all && \
    rm -rf /var/cache/yum

USER podman
WORKDIR /home/podman
```
from then on, if I run directly the same command 
```sh
podman run --name=test_hello_tmp11366 --rm -i --pod=pod_test --label io.podman.compose.config-hash=5e36e74a382d7051c6fcb79d725d6225281563a8a995178f4030ef95c901375a --label io.podman.compose.project=test --label io.podman.compose.version=1.2.0 --label PODMAN_SYSTEMD_UNIT=podman-compose@test.service --label com.docker.compose.project=test --label com.docker.compose.project.working_dir=/test --label com.docker.compose.project.config_files=docker-compose.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=hello --network=test_default --network-alias=hello --tty hello
```
I always get the same output: ```Error: setting up Pasta: pasta failed with exit code 1:```
which makes me think this is not a problem with podman-compose but either with podman or (more likely) my setup.

Can somebody give me a hand?

**Comment by Luap99** [comment]
What pasta version are you using?

**Reply by flixman** [reply]
This is what comes with the image:

```sh
sh-5.2$ pasta --version
pasta 0^20240607.g8a83b53-1.fc40.x86_64-pasta
```

**Reply by Luap99** [reply]
In general there should be error message on the next line of the output, so pasta is not printing an error message at all.
Easiest to tell what is wrong is likely to run it with  `strace -f -l strace.txt podman ...` and the check the logs there. Then check in the strace.txt file for errors. You can also upload the file here so I can take a look.

**Reply by flixman** [reply]
I have been able to reproduce the issue on a minimal example:

```sh
podman pod create --name=pod_test --infra=false --share=
podman network create --label io.podman.compose.project=test --label com.docker.compose.project=test test_default
podman create --name=test_hello_1 --pod=pod_test --label io.podman.compose.config-hash=5e36e74a382d7051c6fcb79d725d6225281563a8a995178f4030ef95c901375a --label io.podman.compose.project=test --label io.podman.compose.version=1.2.0 --label PODMAN_SYSTEMD_UNIT=podman-compose@test.service --label com.docker.compose.project=test --label com.docker.compose.project.working_dir=/test --label com.docker.compose.project.config_files=docker-compose.yml --label com.docker.compose.container-number=1 --label com.docker.compose.service=hello --network=test_default --network-alias=hello hello
strace -f -o ./strace.txt podman start -a test_hello_1
```
Produces the strace.txt file that I have attached. Thank you very much for your help!
[strace.txt](https://github.com/user-attachments/files/16073129/strace.txt)

**Reply by Luap99** [reply, marked as answer]
> 95    openat(AT_FDCWD, "/dev/net/tun", O_RDWR|O_NONBLOCK|O_CLOEXEC) = -1 ENOENT (No such file or directory)
> 95    exit_group(1)  


Is the relevant error. Sounds like you do not have /dev/net/tun setup on your host.


--- 

It is not clear to me why pasta doesn't output the error message here, I see that pasta never prints a error in the strace log so it is not podman loosing it.
@sbrivio-rh @dgibson Does that ring a bell?

**Reply by sbrivio-rh** [reply]
> Is the relevant error. Sounds like you do not have /dev/net/tun setup on your host.

...meaning that you probably need to `modprobe tun` (as root) on the host, by the way.

> It is not clear to me why pasta doesn't output the error message here, I see that pasta never prints a error in the strace log so it is not podman loosing it.
> @sbrivio-rh @dgibson Does that ring a bell?

Yes... I think we just [fixed](https://passt.top/passt/commit/?id=afd9cdc9bb48fdff18d6506e6f5051de35490397) it two weeks ago. Fedora-wise, the fix is included in [passt-0^20240624.g1ee2eca-1.fc40](https://bodhi.fedoraproject.org/updates/FEDORA-2024-4632bfa865), which was moved to "stable" two days ago.

**Reply by flixman** [reply]
thank you very much! I have encountered more issues, but are related to using podman on a rootless podman (as in: I think my setup is not as good as should be :P). Thank you!
